### PR TITLE
Decode byte strings in .msg files correctly

### LINKED ISSUE #24: UnicodeDecodeError
I’ve been encoutering messages such as the following when trying to convert several MSG files:
`UnicodeDecodeError: 'utf-8' codec can't decode byte 0x96 in position 933: invalid start byte`
Unfortunately, I can’t share the MSGs, out of privacy concerns.

I noticed that if I change line #321 in outlookmsgfile.py from:
`return value.decode("utf8")`
to:
`return value.decode("latin-1")`

… then those MSGs seem to be processed fine.
I don’t understand the comment on lines 318-320 very well, and I’m not familiar enough with Python to understand the whole code, so I thought I’d just mention this here rather than doing a pull request.

If any more information or testing on those files is needed, please ask.

### PR DESCRIPTION
**Pull request by MartijnVdS**
Non-Unicode strings in .msg files are encoded using an encoding that is defined in a separate message property (PR_INTERNET_CPID for the body, PR_MESSAGE_CODEPAGE for everything else).

The specification says that this property is required, however some real world .msg files do not have it. This is why the decoding code has a fallback to "cp1252" (Windows code page 1252, "Western Europe").

fixes #24

### CODE REVIEW

**Comment by JoshData** [comment]
Thanks! I don't have any test files that seem to have any STRING8 fields (although they have a mix of cp1252, iso8859_1, utf-8, and ascii stored), so I can't totally tell that this works, but it seems good. I pushed a commit to simplify, in my mind, the code a little bit, and to store the fallback codepage as a module field.

Can you test my changes on a file that you have with these issues?

File: outlookmsgfile.py [hunk 1]
```diff
@@ -240,50 +236,52 @@ def parse_properties(properties, is_top_level, container, doc):
...
+  if "PR_MESSAGE_CODEPAGE" in properties and properties['PR_MESSAGE_CODEPAGE'] in code_pages:
+    properties_encoding = code_pages[properties['PR_MESSAGE_CODEPAGE']]
+
+  # Decode all of the remaining properties.
+  for tag_name, (tag_type, value) in raw_properties.items():
+    if isinstance(tag_type, FixedLengthValueLoader): continue # already done, above
 
-    if not decoded:
-      ret[tag_name] = ret[tag_name].decode(message_codepage, errors='replace')
+    encoding = body_encoding if tag_name == "BODY" else properties_encoding
```
Comment (MartijnVdS) [inline outlookmsgfile.py]: This removes the "in-between" fallback from `body_encoding` to `properties_encoding` before the "don't know what else to try" fallback of `cp1252`. Is that intentional?

Comment (MartijnVdS) [reply outlookmsgfile.py]: (the reason that existed is, I have some files that claim to have a `PR_INTERNET_CPID` of `65001` (UTF-8), but where the body is actually encoded in `PR_MESSAGE_CODEPAGE` -- the only way to figure that out is to try, fail and then retry with the other one)

Comment (JoshData) [reply outlookmsgfile.py]: Not intentional, thanks. I force-pushed a second attempt to get it right.

**Comment by MartijnVdS** [comment]
> Thanks! I don't have any test files that seem to have any STRING8 fields (although they have a mix of cp1252, iso8859_1, utf-8, and ascii stored), so I can't totally tell that this works, but it seems good. I pushed a commit to simplify, in my mind, the code a little bit, and to store the fallback codepage as a module field.
> 
> Can you test my changes on a file that you have with these issues?

Looks like the message files I have get decoded correctly.

**Comment by JoshData** [comment]
Great, merging! Thanks for your help. Very cool.

**Pull request merged**